File: python/euler1_2.py
#!/usr/bin/python3
# Euler1 in Python

def euler(n):
    i = 0
    while i < n:
        if i%3==0 or i%5==0:
            yield i
        i += 1

def euler1(n):
    return sum(euler(n))

print ("Euler1 =", euler1(1000))

File: lua/euler1_2.lua
#!/usr/bin/lua5.3
-- Euler1 in Lua

function euler(n, size)
	return (math.floor(size/n)^2 + math.floor(size/n)) * n / 2 
end

function euler1(n)
	return euler(3, n) + euler(5, n) - euler(15, n)
end

print (string.format("Euler1 = %d", euler1(999)))

